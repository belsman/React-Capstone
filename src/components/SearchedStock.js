import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const SearchedStock = ({ stock }) => (
  <div className="card stock-card mb-1">
    <article className="card-body">
      <h4>{stock.symbol}</h4>
      <h6 className="text-truncate">{stock.name}</h6>
      <div className="my-3 py-2">
        <div className="d-flex justify-content-between mb-1">
          <span>Currency</span>
          <span className="h6 text-secondary">
            {stock.currency}
          </span>
        </div>
        <div className="d-flex justify-content-between">
          <span>Exchange</span>
          <span className="h6 text-secondary">
            {stock.stockExchange}
          </span>
        </div>
      </div>
      <Link to={`/stocks/${stock.symbol}`} className="btn btn-outline-info">
        View stock
      </Link>
    </article>
  </div>
);

SearchedStock.propTypes = {
  stock: PropTypes.shape({
    symbol: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    currency: PropTypes.string.isRequired,
    stockExchange: PropTypes.string.isRequired,
    exchangeShortName: PropTypes.string.isRequired,
  }).isRequired,
};

export default SearchedStock;
