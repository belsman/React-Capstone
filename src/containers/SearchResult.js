import React from 'react';
import { useSelector } from 'react-redux';

import SearchedStock from '../components/SearchedStock';

const SearchResult = () => {
  const results = useSelector(state => state.searchResult);
  const { data, searchStatus, searchError } = results;

  let content;

  if (searchStatus === 'loading') {
    content = <div className="loader">Loading...</div>;
  } else if (searchStatus === 'completed') {
    content = data.map(stock => <SearchedStock key={stock.symbol} stock={stock} />);
  } else if (searchStatus === 'failed') {
    content = <div>{searchError}</div>;
  }

  return (
    <section>
      {content || <h2>No matching Result found!</h2>}
    </section>
  );
};

export default SearchResult;
