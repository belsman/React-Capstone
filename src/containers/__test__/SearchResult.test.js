import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

/* eslint-disable no-unused-vars */
import { useDispatch, useSelector } from 'react-redux';

import ClipLoader from 'react-spinners/ClipLoader';
import override from '../../syledComponent';

import SearchResult from '../SearchResult';
import SearchedStock from '../../components/SearchedStock';

jest.mock('../../components/SearchedStock', () => function SearchedStock() {
  return <div>Hello</div>;
});

jest.mock('react-redux', () => ({
  useDispatch(fn) {
    return fn;
  },
  useSelector() {
    return {
      data: [{
        symbol: 'PRAA',
        name: 'PRA Group, Inc.',
        currency: 'USD',
        stockExchange: 'NasdaqGS',
        exchangeShortName: 'NASDAQ',
      },
      {
        symbol: 'PAAS',
        name: 'Pan American Silver Corp.',
        currency: 'USD',
        stockExchange: 'NasdaqGS',
        exchangeShortName: 'NASDAQ',
      },
      ],
      searchStatus: 'completed',
      searchError: null,
    };
  },
}));

jest.mock('../../syledComponent', () => function override() {
  return 'some css to';
});

jest.mock('react-spinners/ClipLoader', () => ({
  ClipLoader() {
    return (
      <div>Spin</div>
    );
  },
}));

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const match = {
  params: {
    stickerName: 'AAPL',
  },
};

it('renders with with the correct stocks-lists based on the store data', () => {
  act(() => {
    render(<SearchResult />, container);
  });

  const stocks = document.querySelector('section.stock-list-container');
  expect(stocks.children).toHaveLength(2);
});

it('should render the expected output', () => {
  act(() => {
    render(<SearchResult />, container);
  });

  expect(
    pretty(container.innerHTML),
  ).toMatchSnapshot();
});
