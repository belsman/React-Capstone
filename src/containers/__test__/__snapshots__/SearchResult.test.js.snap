// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the expected output 1`] = `
"<div class=\\"p-1\\">
  <section class=\\"stock-list-container mt-2\\">
    <div>Hello</div>
    <div>Hello</div>
  </section>
</div>"
`;
